src/Features/Core/Portable/Diagnostics/Analyzers/QualifyMemberAccessDiagnosticAnalyzerBase.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Semantics;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.QualifyMemberAccess
+{
+    internal abstract class QualifyMemberAccessDiagnosticAnalyzerBase<TLanguageKindEnum> : DiagnosticAnalyzer, IBuiltInAnalyzer where TLanguageKindEnum : struct
+    {
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(WorkspacesResources.MemberAccessShouldBeQualified), WorkspacesResources.ResourceManager, typeof(WorkspacesResources));
+
+        private static readonly LocalizableString s_localizableTitleQualifyMembers = new LocalizableResourceString(nameof(FeaturesResources.AddThisOrMeQualification), FeaturesResources.ResourceManager, typeof(FeaturesResources));
+        private static readonly DiagnosticDescriptor s_descriptorQualifyMemberAccess = new DiagnosticDescriptor(IDEDiagnosticIds.AddQualificationDiagnosticId,
+                                                                    s_localizableTitleQualifyMembers,
+                                                                    s_localizableMessage,
+                                                                    DiagnosticCategory.Style,
+                                                                    DiagnosticSeverity.Hidden,
+                                                                    isEnabledByDefault: true,
+                                                                    customTags: DiagnosticCustomTags.Unnecessary);
+
+        private OptionSet _lazyDefaultOptionSet;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_descriptorQualifyMemberAccess);
+
+        protected abstract ImmutableArray<TLanguageKindEnum> GetSupportedSyntaxKinds();
+        protected abstract bool IsAlreadyQualifiedMemberAccess(SyntaxNode node);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterOperationAction(AnalyzeOperation, OperationKind.FieldReferenceExpression, OperationKind.PropertyReferenceExpression, OperationKind.MethodReferenceExpression);
+        }
+
+        private void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var memberReference = (IMemberReferenceExpression)context.Operation;
+            if (memberReference.Instance == null)
+            {
+                return;
+            }
+
+            if (IsAlreadyQualifiedMemberAccess(memberReference.Instance.Syntax))
```

Comment:
Good catch.  I've added `if (memberReference.Instance.Kind != OperationKind.InstanceReferenceExpression) { return; }` because in your case the kind is `ParameterReference` or it could be `LocalReference`, etc.

I've also added tests.  (I'll push the changes shortly to this PR.)